chef/cookbooks/crowbar/templates/default/crowbar-delete-pacemaker-resources.sh.erb

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+#
+# Old pacemaker resources must be deleted in the old setup before they get created from upgraded node
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+UPGRADEDIR=/var/lib/crowbar/upgrade
+mkdir -p "`dirname "$LOGFILE"`"
+exec >>"$LOGFILE" 2>&1
+
+echo "Executing $BASH_SOURCE"
+
+set -x
+
+mkdir -p $UPGRADEDIR
+
+if [[ -f $UPGRADEDIR/pacemaker-resources-delete-ok ]] ; then
+    echo "Pacemaker resources already successfully deleted"
+    exit 0
+fi
+
+rm -f $UPGRADEDIR/pacemaker-resources-delete-failed
+
+<% if @use_ha %>
+
+echo "Stopping pacemaker resources..."
+
+# Services need to be stopped before deletion.
+# Otherwise we could have services starting up in unwanted locations while deleting constraints.
+
+for type in clone group ms primitive; do
+    for resource in $(crm configure show | awk "\$1 == \"$type\" && ! (\$2 ~ /drbd|stonith|vip-/) {print \$2}"); do
+        crm --wait resource stop $resource
+    done
+done
+
+ret=$?
+if [ $ret != 0 ] ; then
+    echo "Error occured during stopping resources."
+    echo $ret > $UPGRADEDIR/pacemaker-resources-delete-failed
+    exit $ret
+fi
+
+echo "Deleting pacemaker resources..."
+
+crm_cmd=""
+for type in location clone group ms primitive; do
+    for resource in $(crm configure show | awk "\$1 == \"$type\" && ! (\$2 ~ /drbd|stonith|vip-/) {print \$2}"); do
+        crm configure delete $resource
```

Comment:
Shouldn't this line be dropped if you're doing it in a single transaction?

Also I think you could achieve the same thing without the inner `for` loop, e.g. (untested)

    crm_cmd="$crm_cmd"$( crm configure show | awk "\$1 == \"$type\" && ! (\$2 ~ /drbd|stonith|vip-/) {print \"delete \" \$2}" )